downloader/src/main/kotlin/VcsHost.kt

```diff
@@ -161,6 +175,13 @@ enum class VcsHost(
 
         override fun toPermalinkInternal(vcsInfo: VcsInfo, startLine: Int, endLine: Int) =
             toGitPermalink(URI(vcsInfo.url), vcsInfo.revision, vcsInfo.path, startLine, endLine, "#L", "-")
+
+        override fun getArchiveDownloadUrlInternal(vcsInfo: VcsInfo): String? {
+            val projectUrl = URI(vcsInfo.url.removeSuffix(".git"))
+            return projectUrlToUserOrOrgAndProject(projectUrl)?.let {
+                "https://gitlab.com/api/v4/projects/${vcsInfo.url}/repository/archive?ref=${vcsInfo.revision}"
```

Comment:
I believe the more generic GitLab URL (that does not refer to a specific API version) is something like "https://gitlab.com/mbunkus/mkvtoolnix/-/archive/ec80478f87f1941fe52f15c5f4fa7ee6a70d7006/mkvtoolnix-ec80478f87f1941fe52f15c5f4fa7ee6a70d7006.tar.bz2" (and I'd default to bz2 for best compression).

Reply:
No longer relevant